lib/class-wp-rest-menu-items-controller.php

```diff
@@ -446,6 +446,7 @@ protected function prepare_item_for_database( $request ) {
 					} else {
 						$prepared_nav_item['menu-item-position'] = count( $menu_items );
 					}
+					array_push($menu_items,$last_item);
 				} else {
```

Comment:
I had to add this in order to be able to save sub-menu items nested under newly-created items. The problem is that the last item is being removed from $menu_items on line 443, and $menu_items is then used to create a $menu_item_ids array that is used to check if an id is a valid parent. The id of the last saved item will never appear in that array because it was popped out of $menu_items.

This solution feels a bit hacky though; ideas for improvement appreciated 🙂

Reply:
The question is: does `$menu_items` at line 447 need its last item removed? If not, we can get the last item without removing it from the array:

```PHP
$last_item = $menu_items[count($menu_items)-1];
```

and use it on line 445, and if we need one less item on line 447 (for some reason) we'd be better off subtracting 1 there.

I wouldn't say your solution is hacky, just that maybe `array_pop` followed by an `array_push` has the risk of new code appearing between them which is unaware of these updates to `$menu_iteems`.

Reply:
Thanks @draganescu , that's definitely better! We do need one less item on line 447, but agree it's better to subtract 1.